Escape curly brackets in FxGraphDrawer _typename

SungMinCho opened the pull request:
Summary:
Encountered `Error: bad label format` from dot (i.e. graphviz) when benchmarking models that have dict-like structure.

The root cause was that curly brackets were not properly escaped, like this example P522499127 (unescaped curly brackets in target= string)

This diff insert the fix in FxGraphDrawer, since many of these graph generation codes rely on that class.

(Modified summary before exporting to GitHub PR)

Test Plan:
```
CUDA_VISIBLE_DEVICES=7 buck run mode/opt -c python.package_style=inplace //hpc/new/models/feed/benchmark:feed_lower_benchmark -- --model-name={INSERT IFR QE MODEL NAME HERE} --batch-iter 100 --batch-size 768 --num-gpu 1 --lower-presets {INSERT ITS PRESET}
```

Will not encounter dot errors after this diff.

(Modified test plan before exporting to GitHub PR)

Reviewed By: yinghai

Differential Revision: D38758827

jamesr66a commented on torch/fx/passes/graph_drawer.py:
```diff
@@ -140,12 +140,15 @@ def _get_leaf_node(
+                ret = _get_qualified_name(target)
 
-            return _get_qualified_name(target)
+            # Escape "{" and "}" to prevent dot files like P522499127
```
This paste is not accessible to OSS users. Please move it to e.g. a github gist

SungMinCho replied:
Updated it to github gist!

Commit:
Escape curly brackets in FxGraphDrawer _typename (#83604)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83604

Encountered `Error: bad label format` from dot (i.e. graphviz) when benchmarking models that have dict-like structure.

The root cause was that curly brackets were not properly escaped, like this example P522499127 (unescaped curly brackets in target= string)

This diff insert the fix in FxGraphDrawer, since many of these graph generation codes rely on that class.

(Modified summary before exporting to GitHub PR)

Test Plan:
```
CUDA_VISIBLE_DEVICES=7 buck run mode/opt -c python.package_style=inplace //hpc/new/models/feed/benchmark:feed_lower_benchmark -- --model-name={INSERT IFR QE MODEL NAME HERE} --batch-iter 100 --batch-size 768 --num-gpu 1 --lower-presets {INSERT ITS PRESET}
```

Will not encounter dot errors after this diff.

(Modified test plan before exporting to GitHub PR)

Reviewed By: yinghai, jianyuh

Differential Revision: D38758827

fbshipit-source-id: d5d1563181dce2fec446b545cf41cf4ef40a03b8

SungMinCho commented:
@pytorchbot merge

SungMinCho commented:
@pytorchbot merge --force

SungMinCho commented:
@pytorchbot merge --force "force merging, permitted by Sergii"
